src/overlays/actors/ovl_En_Karebaba/z_en_karebaba.c

```diff
@@ -9,67 +13,517 @@ void EnKarebaba_Destroy(Actor* thisx, GlobalContext* globalCtx);
+        rotation.x = -20.0f * Math_Coss(this->actor.shape.rot.x) * Math_Sins(this->actor.shape.rot.y);
+        rotation.y = -20.0f * Math_Coss(this->actor.shape.rot.x) * Math_Coss(this->actor.shape.rot.y);
+
+        for (i = 0; i < 4; i++) {
+            func_800286CC(globalCtx, &position, &sVecZero1, &sVecZero1, 500, 50);
+            position.x += rotation.x;
+            position.y += rotation.z;
+            position.z += rotation.y;
+        };
+
+        func_800286CC(globalCtx, &this->actor.initPosRot.pos, &sVecZero1, &sVecZero1, 500, 100);
+        EnKarebaba_SetupDeadItemDrop(this, globalCtx);
+    }
+}
+
+void EnKarebaba_DeadItemDrop(EnKarebaba* this, GlobalContext* globalCtx) {
+    if (this->actor.params != 0) {
+        this->actor.params--;
+    }
+
+    if (Actor_HasParent(&this->actor, globalCtx) || this->actor.params == 0) {
+        EnKarebaba_SetupDead(this);
+        return;
+    }
+
+    func_8002F554(&this->actor, globalCtx, GI_STICKS_1);
+}
+
+void EnKarebaba_Retract(EnKarebaba* this, GlobalContext* globalCtx) {
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    Math_ApproxF(&this->actor.scale.x, 0.005f, 0.0005f);
+    this->actor.scale.y = this->actor.scale.z = this->actor.scale.x;
+
+    if (Math_ApproxF(&this->actor.posRot.pos.y, this->actor.initPosRot.pos.y + 14.0f, 5.0f)) {
+        EnKarebaba_SetupIdle(this);
+    }
+
+    this->actor.shape.rot.y += 0x1999;
+    EffectSsHahen_SpawnBurst(globalCtx, &this->actor.initPosRot.pos, 3.0f, 0, 12, 5, 1, -1, 10, 0);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Karebaba/func_80A954C0.s")
+void EnKarebaba_Dead(EnKarebaba* this, GlobalContext* globalCtx) {
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Karebaba/func_80A954F8.s")
+    if (this->actor.params != 0) {
+        this->actor.params--;
+    }
+    if (this->actor.params == 0) {
+        EnKarebaba_SetupRegrow(this);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Karebaba/func_80A95548.s")
+void EnKarebaba_Regrow(EnKarebaba* this, GlobalContext* globalCtx) {
+    f32 tmp;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Karebaba/func_80A9559C.s")
+    this->actor.params++;
+    tmp = this->actor.params * 0.05f;
+    Actor_SetScale(&this->actor, 0.005f * tmp);
+    this->actor.posRot.pos.y = this->actor.initPosRot.pos.y + (14.0f * tmp);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Karebaba/func_80A95618.s")
+    if (this->actor.params == 20) {
+        this->actor.flags &= ~0x10;
+        this->actor.flags |= 5;
+        Actor_ChangeType(globalCtx, &globalCtx->actorCtx, &this->actor, ACTORTYPE_ENEMY);
+        EnKarebaba_SetupIdle(this);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Karebaba/func_80A956B8.s")
+void EnKarebaba_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnKarebaba* this = THIS;
+    f32 temp;
+    s32 unused;
```

Comment:
```suggestion
    s32 pad;
```
if required to match, such entirely unused variables are called pad/pad1/pad2...